FirebaseSegmentation/Sources/SEGContentManager.m

```diff
@@ -63,69 +63,112 @@ - (instancetype)initWithDatabaseManager:databaseManager networkManager:networkMa
-      NSString *errorMessage = @"Error getting instance ID.";
-      if (error) {
-        errorMessage = [errorMessage stringByAppendingString:error.description];
-      }
-      NSDictionary *errorDictionary = @{kErrorDescription : errorMessage};
-      completionHandler(NO, errorDictionary);
-      return;
+- (void)associateInstallationWithToken:(FIRInstallationsAuthTokenResult *__nullable)tokenResult
+                  customizedIdentifier:(NSString *)customInstallationID
+                           firebaseApp:(NSString *)firebaseApp
+                                 error:(NSError *_Nullable)error
+                            completion:(SEGRequestCompletion)completionHandler {
+  if (!tokenResult || error) {
+    NSString *errorMessage = @"Error getting AuthToken.";
+    if (error) {
+      errorMessage = [errorMessage stringByAppendingString:error.description];
     }
-
-    strongSelf->_instanceIdentifier = result.instanceID;
-    strongSelf->_instanceIdentifierToken = result.token;
-
-    NSMutableDictionary<NSString *, NSString *> *appAssociationData =
-        [[NSMutableDictionary alloc] init];
-    [appAssociationData setObject:customInstallationID forKey:kSEGCustomInstallationIdentifierKey];
-    [appAssociationData setObject:self->_instanceIdentifier
-                           forKey:kSEGFirebaseInstallationIdentifierKey];
-    [appAssociationData setObject:kSEGAssociationStatusPending forKey:kSEGAssociationStatusKey];
-    [strongSelf->_associationData setObject:appAssociationData forKey:firebaseApp];
-
-    // Update the database async.
-    // TODO(mandard) The database write and corresponding completion handler needs to be wired up
-    // once we support listening to FID changes.
-    [strongSelf->_databaseManager insertMainTableApplicationNamed:firebaseApp
-                                         customInstanceIdentifier:customInstallationID
-                                       firebaseInstanceIdentifier:strongSelf->_instanceIdentifier
-                                                associationStatus:kSEGAssociationStatusPending
-                                                completionHandler:nil];
-
-    // Send the change up to the backend. Also add the token.
-
-    [strongSelf->_networkManager
-        makeAssociationRequestToBackendWithData:appAssociationData
-                                          token:strongSelf->_instanceIdentifierToken
-                                     completion:^(BOOL status,
-                                                  NSDictionary<NSString *, id> *result) {
-                                       // TODO...log, update database.
-
-                                       completionHandler(status, result);
-                                     }];
-  }];
+    NSDictionary *errorDictionary = @{kErrorDescription : errorMessage};
+    completionHandler(NO, errorDictionary);
+    return;
+  }
+  self->_installationIdentifierToken = tokenResult.authToken;
```

Comment:
Just `_installationIdentifierToken` can be used outside of a block. Adding explicit `self->` is required only in the blocks.

This is valid for other instance variables in the method.

Reply:
Thank you!